build/lib/common.sh

```diff
@@ -188,6 +188,30 @@ function build::non-golang::copy_licenses(){
   done
 }
 
+function build::non-golang::generate_attribution(){
```

Comment:
Im not sure this is going to fly honestly.  Worth a shot for sure, but we'll need to get this generated and then create the proper oss ticket to review it and make sure it is complete enough.

Reply:
Looking at the resulting file it does seem reasonable tho.  If we get this approved for redis, we should circle back and add this to the source-controller build which gathers nongolang licenses as well and get them added into our attribution.

Reply:
The format at least is similar to what `generate-attribution` creates.

Redis is odd in that it has snapshots of dependencies in the repo, so this works as it is (kind of). We really need a list of repos and git tags to gather licenses. My only idea is we manually create the file and use it as input along with this script.